kartothek/serialization/_parquet.py

```diff
@@ -405,7 +401,12 @@ def _predicate_accepts(predicate, row_meta, arrow_schema, parquet_reader):
     elif op == ">":
         return max_value > val
     elif op == "in":
-        return bool(val) and not ((max_value < val[0]) or (min_value > val[1]))
+        # We accept the predicate if there is any value in the provided array which is equal to or between
+        # the parquet min and max statistics. Otherwise, it is rejected.
+        for x in val:
+            if min_value <= x <= max_value:
```

Comment:
Maybe try to convert x to a `np.array` and calculate `any( (x_arr >= min_value) & (x_arr <= max_value) )` instead of doing the for loop. This might soften the blow for large `in` predicates.